### PR TITLE
Refactoring `mirrord-layer` `select!` body into functions.

### PR DESCRIPTION
**Pull request by meowjesty**
Still not the best solution, as error handling is (mostly) missing, but I believe this will be a quality of life improvement.

### CODE REVIEW

File: mirrord-layer/src/lib.rs [hunk 1]
```diff
@@ -97,110 +94,129 @@ fn init() {
...
+        }
+        DaemonMessage::Close => todo!(),
+        DaemonMessage::LogMessage(_) => todo!(),
+    }
+}
+
 async fn poll_agent(mut pf: Portforwarder, mut receiver: Receiver<HookMessage>) {
     let port = pf.take_stream(61337).unwrap(); // TODO: Make port configurable
+
+    // NOTE(alex): `codec` is used to retrieve messages from the daemon (messages that are sent
```
Comment (aviramha) [inline mirrord-layer/src/lib.rs:208]: docstrings are welcome, but no need to put names ;P

**Review by aviramha — APPROVED** [review]

**Comment by aviramha** [comment]
Thanks for the needed refactor!

File: mirrord-layer/src/lib.rs (hunk 1 above)
Comment (meowjesty) [reply mirrord-layer/src/lib.rs:208]: Oops, sorry I use an extension to show docs in a tree like view, so tagging them like so helps me with finding stuff.

Do I drop only the `(alex)` part or do you want me to drop the tag (`NOTE`) as well?

Comment (aviramha) [reply mirrord-layer/src/lib.rs:208]: Yeah I assumed that's some sort of an extension. 
I think it'd be better to drop the tag as well, as most developers probably don't share the same extension and it's pretty implicit that a comment is a note ;p

**Review by eyalb181 — APPROVED** [review]

File: mirrord-layer/src/lib.rs (hunk 1 above)
Comment (meowjesty) [reply mirrord-layer/src/lib.rs:208]: Alright, will drop those.

File: mirrord-layer/src/lib.rs [hunk 2]
```diff
@@ -97,110 +94,129 @@ fn init() {
...
     unsafe {
         HOOK_SENDER = Some(sender);
     };
+
     enable_hooks();
+
     RUNTIME.spawn(poll_agent(pf, receiver));
 }
 
+async fn handle_hook_message(
```
Comment (aviramha) [inline mirrord-layer/src/lib.rs:119]: nitpick: add inline, compiler will probably do it automatically, but a hint in the direction is always good (one call site, this function is separated from caller only to be more readable. 
same for the other handle function.

**Review by aviramha — APPROVED** [review]
Approved, check the nitpick and I'll merge it.

File: mirrord-layer/src/lib.rs (hunk 2 above)
Comment (meowjesty) [reply mirrord-layer/src/lib.rs:120]: Done.

**Pull request merged**